zerver/lib/test_helpers.py

```diff
@@ -526,6 +526,9 @@ def find_pattern(pattern: Any, prefixes: list[str]) -> None:
         untested_patterns = {p.replace("\\", "") for p in pattern_cnt if pattern_cnt[p] == 0}
 
         exempt_patterns = {
+            # TODO: Add tests when we are sure about what needs to be
+            # served to the user as summary.
+            "api/v1/messages/summary",
```

Comment:
I think we can write tests now. I expect it should be easy to exercise the API, using the `responses` module to capture the API request that we'd be sending to the model and providing a fixture response that can then verify our own code does what's expected with the response:
- Working with the token counts to adjust the LoggingCountStat correctly.
- Formatting whatever text was in the sample response to send to the client.

Doing that will be helpful for allowing us to verify changes to the logic over time.

Reply:
Okay, makes sense, will do.